Feature/header footer

MathisHTW opened the pull request:
Footer und Header sind fertig für Abgabe leichtes Feature.
Manche Links müssen noch abgepasst werden mit dem Rest des Teams (Siehe Header und Registration/LogIn).

https://dripbox.herokuapp.com/agb

(Gruppe 1, Mathis Schott)

Mathis-ls committed:
Fügt die zwei Seiten /agb und /datenschutz hinzu

Mathis-ls committed:
adds footer layout

Mathis-ls committed:
Merge branch 'main' of https://github.com/Ktechen/Dripbox into feature/header-footer

Mathis-ls committed:
adds AGB Inhalt und Über Inhalt

Mathis-ls committed:
adds Navbar with Logo and 3 Menu options

Mathis-ls committed:
remodells navbar

Mathis-ls committed:
moves navigation to the right

Mathis-ls committed:
adds bootstrap javascript for dropdown menu

Mathis-ls committed:
turns navbar dark

Mathis-ls committed:
makes navbar sticky

Mathis-ls committed:
adds small changes to the header
position of buttons
and names of buttons

Mathis-ls committed:
changes header icon to local data and adds Name of Project

Mathis-ls committed:
removes the display of the header in Log In and Registration screen, when the fitting url is added

viktorianer commented:
Wer, welche Gruppe, das übliche halt ... 💯

viktorianer commented on app/views/layouts/__footer.html.erb:
```diff
@@ -0,0 +1,34 @@
+
+          <%=link_to "AGB",agb_url ,class: "center"%> <br>
+          <%=link_to "Datenschutz", datenschutz_url, class: "center"%> <br>
+          <%=link_to "Über","/uber", class: "center"%>
```
Hier bitte den URL-Helper benutzen, wie in Zeilen davor.

viktorianer commented on app/views/layouts/__footer.html.erb:
```diff
@@ -0,0 +1,34 @@
+
+      <div class="col">
+
+          <%=link_to "AGB",agb_url ,class: "center"%> <br>
```
Style: nach dem Komma bitte Leerzeichen benutzen.
Link: auf der eigenen Seite immer Path-Helper nutzen. Hier also statt `agb_url` dann `agb_path`.

viktorianer commented on app/views/layouts/__header.html.erb:
```diff
@@ -0,0 +1,48 @@
+<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
```
Die Libs zu denen man Abhängigkeiten hat, bitte zentral über Rails-Pipeline und application-Files (.erb, .scss, .js) einbinden. 
Keinen direkten Aufruf von externen Libs, schon gar nicht pro einzelne Seiten.

https://edgeguides.rubyonrails.org/asset_pipeline.html

MathisHTW replied:
ich verstehe dass das nicht in die Headerdatei soll, aber nicht die rails doku zur pipeline ... 
soll ich das in vendor/assets/javascript packen und dann irgendwie in den application header rendern?

viktorianer replied:
Du musst Bootstrap 4 oder besser 5 über Webpacker ausliefern und über Yarn installieren. Schaue dazu online Tutorials an. Sonst schauen wir es Montag noch mal an.

viktorianer commented on app/views/layouts/__header.html.erb:
```diff
@@ -0,0 +1,48 @@
+
+      </ul>
+
+      <!-- Für mittleres feature -->
+      <!--
+      <form class="form-inline my-2 my-lg-0">
+        <input class="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search">
+        <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
+      </form>
+      -->
```
Keine auskommentierten Code für die "Zukunft". Entweder richtig nutzen oder entfernen.

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -2,15 +2,24 @@
   </head>
 
   <body>
+    <%unless current_page?('/')%><!-- Header soll nicht in Registration Log in auftauchen -> Links für die beiden Seiten hier noch einzufügen-->
+      <%=render "layouts/_header" %>
+    <%end%>
```
Dafür gibt es bereits ein Feature von Rails. Das baust du gerade hier nach.

https://edgeguides.rubyonrails.org/layouts_and_rendering.html#using-the-content-for-method

MathisHTW replied:
ich weiß nicht was hier gemeint ist. das unless oder das rendern
und es sind doch nur 2 zeilen code. Wieso soll ich dafür unbedingt ein Rails Feature nutzen?

viktorianer replied:
Weil man das Framework lernen sollte und nicht alles direkt falsch machen. Um es genau zu sagen :).
Heute 2 Zeilen, morgen noch 2. Am Ende ist das ganze Projekt nicht mehr wartbar. Vor allem dann nicht, wenn jemand dann reinkommt, der Rails gut kennt. Dann versteht er nichts mehr. Er wartet, dass die Features von Rails benutzt werden.

viktorianer replied:
Gemeint ist immer das, was in der Nachricht oben zu sehen ist. Also alle 3 Zeilen.

viktorianer commented on app/views/site_informations/about.html.erb:
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Title</title>
+</head>
+<body class = "backroundcolorAbout">
```
Codeverdopplung, da bereits durch `app/views/layouts/application.html.erb` gegeben. In den Views brauchst du kein `html.head`, `body`, etc. einfügen.

https://edgeguides.rubyonrails.org/layouts_and_rendering.html#understanding-yield

viktorianer commented on app/views/site_informations/agb.html.erb:
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Title</title>
+</head>
+<body>
```
Dasselbe wie in `app/views/site_informations/about.html.erb`

viktorianer commented on app/views/site_informations/datenschutz.html.erb:
```diff
@@ -0,0 +1,750 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Title</title>
+</head>
+<body>
```
Dasselbe wie in `app/views/site_informations/about.html.erb`

viktorianer commented on config/routes.rb:
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  get '/agb', to:"site_informations#agb"
+  get '/datenschutz', to:"site_informations#datenschutz"
+  get '/uber',to:"site_informations#about"
+  get '/ueber',to:"site_informations#about"
```
Bitte auf Formatierung achten. Das kann man ja kaum lesen :)

viktorianer reviewed (changes requested):
Die Seiten sind relativ schlicht, kaum CSS oder HTML. Etwas mehr Aufwand wäre besser für 💯 .

viktorianer commented:
Bitte ladet immer alle Teammitglieder ein! Wie in den Folien es steht und wie in der Vorlesung besprochen.

Mathis-ls committed:
1st pull request change

Mathis-ls committed:
2. pull request change

Mathis-ls committed:
3rd request: moved library into application.html.erb

Mathis-ls committed:
4th request: deleted comment

viktorianer commented:
Bitte bei den Commit Messages immer einen sinnvollen Text verwendet, der andeutet, was dieser Commit macht. z.B.: `Use routes helper in views` oder `Replace *_url by *_path routes helper`

viktorianer commented on app/views/layouts/__header.html.erb:
```diff
@@ -0,0 +1,37 @@
+            Data
+          </a>
+          <div class="dropdown-menu" aria-labelledby="navbarDropdown">
+            <a class="dropdown-item" href="#">Dokumente</a> <!-- Link für eigene Dokumentseite -->
+            <a class="dropdown-item" href="#">Bilder</a> <!-- Link für eigene Bilderseite -->
+            <div class="dropdown-divider"></div>
+            <a class="dropdown-item" href="#">Für mich freigegeben</a> <!-- Link für Seite, die alle für mich freigegebenen Dateien  -->
```
Bitte `link_to` Viewhelper bentuzen, statt alles in HTML selbst schreiben. Das hat den Vorteil, dass Rails darauf Einfluss nehmen kann, wo es muss, und damit weniger Aufwand bei späteren Features und Änderungen entsteht.

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -1,16 +1,29 @@
 <html>
   <head>
+
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
```
Rails Pipeline nutzen.

viktorianer replied:
Ahja, das ist doppelter Code, weil bereits im Header drin.

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -1,16 +1,29 @@
   </head>
 
   <body>
+    <%unless current_page?('/')%><!-- Header soll nicht in Registration Log in auftauchen -> Links für die beiden Seiten hier noch einzufügen-->
```
Auch hier Rails Helper nutzen. Statt `current_page?('/')` lieber `current_page?(root_path)`.

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -1,16 +1,29 @@
+      <%=render "layouts/_header" %>
+    <%end%>
     <%= yield %>
+    <%=render "layouts/_footer" %>
```
Hier besser `<%= render 'layouts/footer'%>`

viktorianer replied:
Du hast die Partials falsch benannt. Da muss nur ein `_` im Dateinamen stehen.

Mathis-ls committed:
request 6: removed unnecessary code

Mathis-ls committed:
request 7: removed unnecessary code

Mathis-ls committed:
request 8: removed unnecessary code

Mathis-ls committed:
request 9: changed formatting in routes

Mathis-ls committed:
changes check for current site in application.html.erb, so that it is now using rails given
content_for method with named yield.
Every erb file now has to insert <% content_for :header do%>
                                   <%=render "layouts/_header" %>
                                 <%end %>

for the header to load

Mathis-ls committed:
changes raw registraition url into rails path variable

Mathis-ls committed:
refactors names of partials and changes "" into '' int application.html.erb for layouts/footer

Mathis-ls committed:
some changes from href to link_to in header

viktorianer reviewed (approved):
Das schaut schon viel besser aus 👍🏼 
Bleibt nur noch Bootstrapeinbindung über Assets.
Gebe aber schon mal frei, damit du gleich weiter machen kannst, wenn du das behoben hast.

Mathis-ls committed:
adds beautiful changes to html/css on small sites

viktorianer commented:
> Wer, welche Gruppe, das übliche halt ... 💯

Kannst du noch bitte das hier erledigen? Danke!

notefox reviewed (approved):
Werf's durch wenn alles läuft ^^

NoteFox commented:
> > Wer, welche Gruppe, das übliche halt ... 💯
> 
> Kannst du noch bitte das hier erledigen? Danke!

Ist drin

Ktechen commented:
Merge mal

MathisHTW committed:
Merge branch 'main' into feature/header-footer

MathisHTW commented:
damn son
